safenode/src/storage/chunks.rs

```diff
@@ -86,4 +64,30 @@ impl ChunkStorage {
 
         Ok(())
     }
+
+    #[allow(dead_code)]
+    pub(super) async fn addrs(&self) -> Vec<ChunkAddress> {
+        self.cache
+            .read()
+            .await
+            .iter()
+            .map(|(addr, _)| *addr)
+            .collect()
+    }
+
+    #[allow(dead_code)]
+    pub(super) async fn remove_chunk(&self, address: &ChunkAddress) -> Result<()> {
+        trace!("Removing Chunk: {address:?}");
```

Comment:
Because when a node responsibility-range shrinks so that chunks fall out of it, they can optionally remove chunks if they are getting close to the minimum required space.

Reply:
Shouldn't they stop storing new chunks instead?

Reply:
No, because the new chunks _are_ within their responsibility range. The chunks that fell outside that range as their range shrinks, are _not_ their responsibility.

The former must be served, or they will be punished. The latter are not expected from them.

Reply:
You know, when the network grows, every node has a smaller and smaller range that they are responsible for.

The first node is responsible for the entire range.

So, when network grows, the responsibility range shrinks.

Any chunk that falls outside the range, can be dropped.